src/main/java/com/synopsys/integration/alert/web/config/JobConfigController.java

```diff
@@ -117,10 +121,8 @@ public JobConfigController(final JobConfigActions jobConfigActions, final Respon
 
         if (optionalModel.isPresent()) {
             final JobFieldModel fieldModel = optionalModel.get();
-            final boolean missingPermission = fieldModel.getFieldModels().stream()
-                                                  .map(model -> AuthorizationManager.generatePermissionKey(model.getContext(), model.getDescriptorName()))
-                                                  .anyMatch(permissionKey -> !authorizationManager.hasReadPermission(permissionKey));
-            if (missingPermission) {
+            final boolean hasPermissions = fieldModel.getFieldModels().stream().allMatch(model -> authorizationManager.hasReadPermission(model.getContext(), model.getDescriptorName()));
```

Comment:
These are used in multiple places. We should create a new method that is passed a function to do the specific permission check.